### ISSUE TITLE
Highlighting and improving Transfer use cases

### ISSUE THREAD

**Issue by FALLAI-Denis** [IBM/dbb-zappbuild#251]
Hi,

The zAppBuild model provides for the management of the types of elements to be made available and packaged without transformation: `Transfer.properties` and `Transfer.groovy`.

As far as we are concerned, the provision of elements without transformation represents 80% to 90% of our element types, while elements with transformation, essentially programs, represent more than 80% of the elements occurrences. What matters for an IBM DBB administrator is the number of element types, and their differents transformation processes, rather than the number of elements occurrences.

The Transfer processing type should be highlighted in the same way as the other processing types.

In addition, all item types with Transfer processing are not necessarily sources in LRECL 80 format, and some are not eligible for PDS handling.
The `Transfer.properties` and `Transfer.groovy` models should provide a processing case using a PDS other than LRECL 80, and a processing case using a sequential file (not PDS).

Thanks.

**Comment by dennis-behm** [comment]
Hi @FALLAI-Denis ,

thanks for providing feedback on the implementation of the `Transfer` element type. When you refer to highlighting the Transfer element type - are you looking for some samples? Have you checked out the sample in https://github.com/IBM/dbb-zappbuild/main/samples/application-conf/file.properties#L39-L48

To your idea about supporting additional datasets characteristics: 

I think an additional PropertyMapping probably satisfies your needs to define the dataset characteristics, same how it used in . However, I feel, that the configuration would be a bit clumsy.

What do you think about such a configuration for control cards. The last setting would allow to map files to the dataset characteristics:

```
#
# PropertyMapping to map files using the Transfer.groovy language script to different target datasets 
#
transfer_datasetMapping = transfer_jclPDS :: **/*.jcl
transfer_datasetMapping = transfer_controlCardPDS :: **/*.ccard
#
# file mapping for overwriting the default deployType of the Transfer.groovy language script 
#
transfer_deployType = JCL :: **/*.jcl
transfer_deployType = CONTROLCARD :: **/*.ccard
#
# --> dataset characteristics mapping
transfer_datasetOptions= transfer_srcOptions :: **/*.jcl
transfer_datasetOptions= transfer_controlCardOptions :: **/*.ccard
```
An additional new setting is required in build-conf/Transfer.properties
```
#
# dataset creation options
transfer_srcOptions=cyl space(1,1) lrecl(80) dsorg(PO) recfm(F,B) dsntype(library)
transfer_controlCardOptions=cyl space(1,1) lrecl(133) dsorg(PO) recfm(F,B) dsntype(library)
```
Based on those single assignments of the property values, a user might (unfortunately) incorrectly map a file, which might lead to truncation and an exception. But this might be happening in your existing setup as well.

What are your thoughts on the above proposal?

Dennis

**Comment by FALLAI-Denis** [comment]
Hi @dennis-behm 

_When you refer to highlighting the Transfer element type - are you looking for some samples?_

Yes, but also indicate it explicitly on the "README.md" of page https://github.com/IBM/dbb-zappbuild/blob/main/languages

**Comment by dennis-behm** [comment]
The transfer script has been enhanced in zAppBuild 3.5.0 https://github.com/IBM/dbb-zappbuild/releases/tag/3.5.0

The management of default build configuration has been implemented as part of zAppBuild 3.4.0 https://github.com/IBM/dbb-zappbuild/releases/tag/3.4.0
